pkg/certificate/manager.go

```diff
@@ -128,6 +130,16 @@ func (m *Manager) EnsureCertificate(certReq Request, ownerID int) (Certificate,
 		config := cli.Config{
 			CAFile:    "file:" + certReq.CACert,
 			CAKeyFile: "file:" + certReq.CAKey,
+			CFG: &cfsslconfig.Config{
+				Signing: &cfsslconfig.Signing{
+					Profiles: map[string]*cfsslconfig.SigningProfile{},
+					Default: &cfsslconfig.SigningProfile{
+						Usage:        []string{"signing", "key encipherment", "server auth", "client auth"},
```

Comment:
What happens if we leave the signing profile and the usages unspecified, here?

Reply:
The certificate issuance will fail, or the certificate will lose certain functionalities. This is using the same default configuration as before, with the only modification to the `Expiry` value:

```go
// DefaultConfig returns a default configuration specifying basic key
// usage and a 1 year expiration time. The key usages chosen are
// signing, key encipherment, client auth and server auth.
func DefaultConfig() *SigningProfile {
	d := helpers.OneYear
	return &SigningProfile{
		Usage:        []string{"signing", "key encipherment", "server auth", "client auth"},
		Expiry:       d,
		ExpiryString: "8760h",
	}
}
```

https://github.com/cloudflare/cfssl/blob/ed8df49712657524e8647cb7a9576422baa2faa1/config/config.go#L675-L685